internal/core/cobra_utils.go

```diff
@@ -78,6 +83,63 @@ func cobraRun(ctx context.Context, cmd *Command) func(*cobra.Command, []string)
 	}
 }
 
+// handlePositionalArg will catch positional argument if command has one.
+// When a positional argument is found it will mutate its value in rawArgs to match the argument unmarshaller format.
+// E.g.: '[value b=true c=1]' will be mutated to '[a=value b=true c=1]'.
+// It returns errors when:
+// - no positional argument is found.
+// - a positional argument exists, but is not positional.
+// - an argument duplicates a positional argument.
+func handlePositionalArg(cmd *Command, rawArgs []string) error {
+	positionalArg := cmd.ArgSpecs.GetPositionalArg()
+
+	// Command does not have a positional argument.
+	if positionalArg == nil {
+		return nil
+	}
+
+	// Positional argument is found condition.
+	positionalArgumentFound := len(rawArgs) > 0 && !strings.Contains(rawArgs[0], "=")
+
+	// Argument exists but is not positional.
+	for i, arg := range rawArgs {
+		if strings.HasPrefix(arg, positionalArg.Prefix()) {
+			return &CliError{
+				Err:  fmt.Errorf("a positional argument is required for this command"),
+				Hint: positionalArgHint(cmd, strings.TrimLeft(arg, positionalArg.Prefix()), append(rawArgs[:i], rawArgs[i+1:]...), positionalArgumentFound),
+			}
```

Comment:
nit, feel free to accept of not.
```suggestion
			argumentValue := strings.TrimLeft(arg, positionalArg.Prefix())
			otherArgs := append(rawArgs[:i], rawArgs[i+1:]...)
			return &CliError{
				Err:  fmt.Errorf("a positional argument is required for this command"),
				Hint: positionalArgHint(cmd, argumentValue, otherArgs, positionalArgumentFound),
			}
```